browser/mac/BUILD.gn

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+check_includes = false
```

Comment:
Did you check to see how bad things are with this removed? (ex: when running `npm run gn_check`). I know we try to avoid adding new instances of this. But maybe a follow up is a better place to address

Reply:
That's a good point, I forgot I had to do that. Unfortunately, without that `gn_check` failed like this and I wasn't too sure how to get around that:

```
$ npm run gn_check Static --channel=nightly --target_arch=x64

> brave-core@1.64.51 gn_check
> node ./build/commands/scripts/commands.js gn_check Static

-----------------------------------------------------------------------------------------------------------------------
/Users/emerick/work/cr122/brave-browser/src
> gn check /Users/emerick/work/cr122/brave-browser/src/out/Static
ERROR at //brave/browser/mac/keystone_glue.mm:43:11: Include not allowed.
#include "brave/browser/mac/sparkle_glue.h"
          ^-------------------------------
It is not in any dependency of
  //brave/browser/mac:keystone_glue
The include file is in the target(s):
  //chrome/browser:browser
which should somehow be reachable.
```

Maybe @bridiver or @goodov have some advice here?

Reply:
Per DM, bad things happen with this removed! Thanks for helping share impact 😄 

Nothing to see here
![homer](https://github.com/brave/brave-core/assets/4733304/6c56d99b-2754-43f5-b767-dd866f582d03)

Reply:
That makes sense. I believe I tried that and it got a little complicated. I think given the timing with this release and the testing required for the updater, we'll have to address this one in a follow-up issue. I'll enter that issue now and will assign to myself.

Reply:
https://github.com/brave/brave-browser/issues/35893